Lab1

gudkovst opened the pull request.

gudkovst committed:
Add files via upload

gudkovst committed:
application for discovering copies on local network

gudkovst committed:
application for discovering copies on local network

gudkovst committed:
application for discovering copies on local network

gudkovst committed:
Create README.md

gudkovst committed:
Update README.md

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
Rename gradlew to lab1/gradlew

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

Gregpack commented on main/Application.java:
```diff
@@ -0,0 +1,132 @@
+    public static void main(String[] args) throws IOException {
+        if (args.length < 2){
+            System.out.print(InetAddress.getLocalHost().toString());
+            System.err.print("Too few arguments\n");
```
не очень понятное сообщение об ошибке

Gregpack commented on main/Application.java:
```diff
@@ -0,0 +1,132 @@
+import java.net.UnknownHostException;
+import java.util.*;
+
+public class Application {
```
всё в одном классе - не очень хорошая практика. разнеси на несколько классов, пожалуйста

Gregpack replied:
в классы можно вынести то что ты запускаешь в тредах как класс который implements Runnable

Gregpack commented on main/Application.java:
```diff
@@ -0,0 +1,132 @@
+import java.io.IOException;
```
если собираешь градлом, то у тебя структура папок должна быть src/main/java и дальше уже код.

Gregpack commented on main/Application.java:
```diff
@@ -0,0 +1,132 @@
+        multicastSocket = new MulticastSocket(port);
+        multicastSocket.joinGroup(addrMultiGroup);
+        Timer sendTimer = new Timer();
+        sendTimer.schedule(new TimerTask() {
```
лучше используй scheduledThreadPool из Executors класса

Gregpack commented on main/Application.java:
```diff
@@ -0,0 +1,132 @@
+                if (newApplAddr != null)
+                    if (copiesAppl.put(newApplAddr.toString(), System.currentTimeMillis()) == null)
+                        for (String copy : copiesAppl.keySet()) {
+                            if (isValid(copiesAppl.get(copy)))
```
расставь {} скобки, пожалуйста

gudkovst committed:
new structure of project

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
new structure of project

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
new structure of project

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
application for discovering copies on local network

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>

gudkovst committed:
new structure of project

Исправил замечания, сделал красивую многопоточность, задания для потоков выделил в отдельные классы.

Signed-off-by: gudkovst <69453922+gudkovst@users.noreply.github.com>
